xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/VaapiEGL.cpp

```diff
@@ -445,8 +468,279 @@ bool CVaapiTexture::TestInteropHevc(VADisplay vaDpy, EGLDisplay eglDisplay)
+  if (status != VA_STATUS_SUCCESS)
+  {
+    CLog::LogFunction(LOGERROR, "CVaapiTexture::MapEsh", "vaSyncSurface - Error: %s (%d)", vaErrorStr(status), status);
+    return false;
+  }
+
+  m_textureSize.Set(surface.width, surface.height);
+
+  for (int layerNo = 0; layerNo < surface.num_layers; layerNo++)
+  {
+    int plane = 0;
+    auto const& layer = surface.layers[layerNo];
+    if (layer.num_planes != 1)
+    {
+      CLog::LogFunction(LOGDEBUG, "CVaapiTexture::MapEsh", "DRM-exported layer has %d planes - only 1 supported", layer.num_planes);
+      return false;
+    }
+    auto const& object = surface.objects[layer.object_index[plane]];
+
+    MappedTexture* texture{};
+    EGLint width{m_textureSize.Width()};
+    EGLint height{m_textureSize.Height()};
+
+    switch (surface.num_layers)
+    {
+      case 2:
+        switch (layerNo)
+        {
+          case 0:
+            texture = &m_y;
+            break;
+          case 1:
+            texture = &m_vu;
+            if (surface.fourcc == VA_FOURCC_NV12 || surface.fourcc == VA_FOURCC_P010 || surface.fourcc == VA_FOURCC_P016)
+            {
+              // Adjust w/h for 4:2:0 subsampling on UV plane
+              width = (width + 1) >> 1;
+              height = (height + 1) >> 1;
+            }
+            break;
+          default:
+            throw std::logic_error("Impossible layer number");
+        }
+        break;
+      default:
+        CLog::LogFunction(LOGDEBUG, "CVaapiTexture::MapEsh", "DRM-exported surface %d layers - only 2 supported", surface.num_layers);
+        return false;
+    }
+
+    EGLint attribs[13 + 4 /* modifiers */] = {
+      EGL_LINUX_DRM_FOURCC_EXT, static_cast<EGLint> (layer.drm_format),
+      EGL_WIDTH, static_cast<EGLint> (width),
+      EGL_HEIGHT, static_cast<EGLint> (height),
+      EGL_DMA_BUF_PLANE0_FD_EXT, object.fd,
+      EGL_DMA_BUF_PLANE0_OFFSET_EXT, static_cast<EGLint> (layer.offset[plane]),
+      EGL_DMA_BUF_PLANE0_PITCH_EXT, static_cast<EGLint> (layer.pitch[plane]),
+      // When adding parameters, be sure to update the &attribs reference for the
+      // plane modifiers below
+      EGL_NONE
+    };
+    if (m_hasPlaneModifiers)
+    {
+      EGLint* attrib = &attribs[12];
+      *attrib++ = EGL_DMA_BUF_PLANE0_MODIFIER_LO_EXT;
+      *attrib++ = static_cast<EGLint> (object.drm_format_modifier);
+      *attrib++ = EGL_DMA_BUF_PLANE0_MODIFIER_HI_EXT;
+      *attrib++ = static_cast<EGLint> (object.drm_format_modifier >> 32);
+      *attrib++ = EGL_NONE;
+    }
+
+    texture->eglImage = m_interop.eglCreateImageKHR(m_interop.eglDisplay,
+                                                   EGL_NO_CONTEXT, EGL_LINUX_DMA_BUF_EXT, nullptr,
+                                                   attribs);
+    if (!texture->eglImage)
+    {
+      CEGLUtils::LogError("Failed to import VA DRM surface into EGL image");
+      return false;
+    }
+
+    texture->glTexture = ImportImageToTexture(texture->eglImage);
```

Comment:
never create a method that is only called once. better add those 6 lines of code here.

Reply:
It will be called twice once it is moved to the base class and `Vaapi1Texture` also uses it :-)

Reply:
never create methods that are not needed now but maybe in the future

Reply:
Why in the future? I would have done it as part of this PR once the Vaapi2 stuff is OK.

Also, splitting code into appropriately named functions (even if they are not reused) is actually good practice because it makes the code more self-describing and easier to read/test/debug if each function only does one thing. I see that we are very far from that in many parts of the Kodi code, but why should we actively work against it in new code?

Reply:
I don't see the function being used more that once in this diff. hence drop it